### PR TITLE
add a HOC for smoothing count down timer to 1s resolution and make sure the upload banner shows up for at least 2s

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/fs/footer/upload-countdown-hoc.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      */
+          break
+      }
+    }
+
+    _startTicker = () => {
+      if (this._tickerID) {
+        return
+      }
+      this._tickerID = this.props.setInterval(this._tick, tickInterval)
```
Comment (chrisnojima) [inline shared/fs/footer/upload-countdown-hoc.js]: likely want to call this._stopTicker here

File: shared/fs/footer/upload-countdown-hoc.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  class extends React.PureComponent<Props, State> {
+    state = {
+      displayDuration: 0,
+      mode: 'hidden',
+      glueTTL: 0,
+    }
+
+    _tickerID: ?IntervalID = null
+
+    _tick = () => {
```
Comment (chrisnojima) [inline shared/fs/footer/upload-countdown-hoc.js]: you need to protect against unmount here

File: shared/fs/footer/upload-countdown-hoc.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      const {mode, glueTTL, displayDuration} = this.state
```
Comment (chrisnojima) [inline shared/fs/footer/upload-countdown-hoc.js]: this is sort of dangerous, setState is async so you reading it and then calling setState won't necessarily be correct all the time, likely safer to use
```
this.setState(prevState => {}) 
```
form

File: shared/fs/footer/upload-countdown-hoc.js (hunk 2 above)
Comment (songgao) [reply shared/fs/footer/upload-countdown-hoc.js]: I thought the TimerHOC already did that. Is it not enough?

File: shared/fs/footer/upload-countdown-hoc.js (hunk 3 above)
Comment (chrisnojima) [reply shared/fs/footer/upload-countdown-hoc.js]: oh i didn't realize it was composed with that, tnx

File: shared/fs/footer/upload-countdown-hoc.js (hunk 2 above)
Comment (chrisnojima) [reply shared/fs/footer/upload-countdown-hoc.js]: oh i didn't realize it was composed with that, tnx

File: shared/fs/footer/upload-countdown-hoc.js (hunk 1 above)
Comment (songgao) [reply shared/fs/footer/upload-countdown-hoc.js]: I meant this call to be idempotent, i.e. start the ticker if it hasn't been started. The `if` condition above already covers the case where we already have a ticker ID. I'll add a comment to clarify.

**Comment by songgao** [comment]
@chrisnojima ready for another look. Thanks!

**Comment by songgao** [comment]
@chrisnojima ping 🏓

**Pull request merged**